astropy/io/misc/hdf5.py

```diff
@@ -58,7 +58,7 @@ def is_hdf5(origin, filepath, fileobj, *args, **kwargs):
         return isinstance(args[0], (h5py.highlevel.File, h5py.highlevel.Group, h5py.highlevel.Dataset))
 
 
-def read_table_hdf5(input, path=None):
+def read_table_hdf5(input, path=None, **kwargs):
```

Comment:
Use an explicit `character_as_bytes=True` keyword arg here and remove the new `**kwargs`.  That makes the function API be properly documented and obvious.  (And of course update the docstring).